File: src/main/java/com/eric/okpo_java/lab2/Task12.java
package com.eric.okpo_java.lab2;

import java.util.Scanner;

public class Task12 {

    public static void main(String[] args) {
        var input = new Scanner(System.in);
        System.out.print("Введите значения x1, y1, x2, y2: ");

        int x1 = input.nextInt();
        int y1 = input.nextInt();
        int x2 = input.nextInt();
        int y2 = input.nextInt();

        if (x1 == x2 || y1 == y2) {
            System.out.println("Ладья может побудить за один ход.");
        } else {
            System.out.println("Ладья не может победить за один ход.");
        }
    }
}

File: src/main/java/com/eric/okpo_java/lab5/Task12.kt
package com.eric.okpo_java.lab5

class Task12 {

    fun main() {
        val array = DoubleArray(9)

        println("Введите 9 элементов массива array:")
        for (i in 0..8) {
            println("array[$i]: ")
            array[i] = readln().toDouble()
        }

        for (i in 0..8) {
            if (array[i] > 1) array[i] -= 0.5
            else array[i] += 0.5
        }

        println("Измененный массив array:")
        for (i in 0..8) {
            println("array[$i]: ${array[i]}")
        }
    }
}